packages/data/src/index.js

```diff
@@ -229,3 +234,24 @@ export const use = defaultRegistry.use;
  * @param {StoreDescriptor} store Store descriptor.
  */
 export const register = defaultRegistry.register;
+
+const experimentalAPIs = __dangerousOptInToUnstableAPIsOnlyForCoreModules(
+	'I know using unstable features means my plugin or theme will inevitably break on the next WordPress release.',
+	'@wordpress/data'
+);
+
+function __experimentalPrivateSelector( { name }, selector ) {
+	return ( ...args ) =>
+		selector( defaultRegistry.stores[ name ].store.getState(), ...args );
```

Comment:
It is a major shortcoming that the selector always selects from the default registry, and not the one that's in React context and is returned by `useRegistry`. I wonder if we can do anything about it?

Reply:
```js
// registration
registerPrivateSelectors( blockEditorStore, {
  getContentLockingParent
} );

// plain usage
const { privateOf } = unlock( dataExperiments );
privateOf( registry.select( blockEditorStore ) ).getContentLockingParent();

// usage in React
useSelect( ( select ) => ( {
  parent: privateOf( select( blockEditorStore ) ).getContentLockingParent();
} ) );
```
I can register private selectors and actions on a store, and then the object returned by `select` has the good old public methods, but can also be "unlocked" with `privateOf` to access the private parts.

The same can be done with actions and `dispatch`.

Ideally, I would not need a custom `privateOf` function, but could use the general purpose `unlock` to unlock any object. Could be implemented by having a `Symbol.for( 'accessKey' )` property on an object. Very similar to how, to a `for of` loop, you can pass either an object with a `Symbol.iterator` field or an actual iterator.

Reply:
@jsnajdr I really like this idea and I'm exploring an implementation. Would you mind taking a look? I think the new lockObj/unlockObj API could very well replace the old register/unlock API – I just need some better names.

Reply:
I'm adding this to my todo-list after finishing the React 18 migration. Which currrently utilizes me 100% 🙂